### PR TITLE
Drop support in GC b&r using distCp

### LINKED ISSUE #4340: Remove distcp GC backup and restore from docs - drop support
If we are keepign b&r, then add basic integration tests
Design the thinnest test possible for this one and estimate the effort. 

Add integration tests for https://github.com/treeverse/lakeFS/blob/master/clients/spark/core/src/main/scala/io/treeverse/clients/GCBackupAndRestore.scala

### PR DESCRIPTION
**Pull request by talSofer**
Closes  #4340

### CODE REVIEW

File: docs/howto/garbage-collection.md [hunk 1]
```diff
@@ -286,9 +286,7 @@ using the feature you may want to first gain confidence in the decisions GC make
 * Backup: copy expired objects from your repository's storage namespace to an external location before running GC in [sweep only mode](#sweep-only-mode).  
 * Restore: copy objects that were hard-deleted by GC from an external location you used for saving your backup into your repository's storage namespace.
 
-### Using rclone
-
-Follow [rclone documentation](https://rclone.org/docs/) to configure remote access to the underlying storage used by lakeFS.
+Follow [**rclone** documentation](https://rclone.org/docs/) to configure remote access to the underlying storage used by lakeFS.
```
Comment (johnnyaug) [inline docs/howto/garbage-collection.md:289]: Related?

**Review by johnnyaug — APPROVED** [review]
LGTM

File: docs/howto/garbage-collection.md (hunk 1 above)
Comment (talSofer) [reply docs/howto/garbage-collection.md:289]: yes, I. only kept a single option that's left therefore removed the title that says rclone or distcp

Comment (talSofer) [reply docs/howto/garbage-collection.md]: ok, I removed it. thanks!

**Review by N-o-Z — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**